derive/src/lib.rs

```diff
@@ -77,7 +78,10 @@ fn expand_contract(args: ContractArgs) -> Result<TokenStream> {
     //   the path will always be rooted on the cargo manifest directory.
     //   Eventually we can use the `Span::source_file` API to have a better
     //   experience.
-    let artifact_path = args.artifact_path;
+    let artifact_path = {
+        let full_path = fs::canonicalize(args.artifact_path.value())?;
+        LitStr::new(&full_path.to_string_lossy(), args.artifact_path.span())
+    };
```

Comment:
Unrelated to the purpose of this PR 🙈 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FB3nLKeJ9wUPbG%2Fgiphy.gif&token=unrelated" data-gitmeme-token="unrelated"><img src="https://media0.giphy.com/media/B3nLKeJ9wUPbG/giphy.gif" title="Created by gitme.me with /unrelated"/></a>

Reply:
Hehe yes it is.

Reply:
@fleupold Want me to move this to a separate PR? I agree it should probably be kept separate.

Reply:
What does the `Lit` stand for in `LitStr`?

Reply:
I think it's fine to keep it in here, but it should become a habit to create small PRs for small fixes. It makes it easier to track what changed when and allows reverting individual features. E.g. if we were to revert this commit, we'd also revert all the great `must use` annotations.